### PR TITLE
feat: implement recurring task scheduler with SQLite deduplication

### PR DESCRIPTION
**Pull request by jessmartin**
## Summary

Implements a comprehensive recurring task scheduling system with SQLite-based deduplication for server-side task execution. This addresses the recurring tasks requirement from the MCP email integration master plan.

- **ProcessedTaskTracker**: SQLite-based deduplication using atomic `INSERT OR IGNORE` pattern
- **TaskScheduler**: Leverages existing AgenticLoop for LLM-powered task execution
- **process-tasks.ts**: Standalone cron job script designed for Render deployment
- **Comprehensive testing**: 33 passing tests covering all edge cases and error scenarios
- **Multi-store support**: Per-store task deduplication with proper isolation
- **Execution windowing**: 10-minute window to catch missed task executions
- **Event emission**: Proper task execution events emitted to LiveStore

### SQLite Deduplication
- Uses atomic `INSERT OR IGNORE` for race-condition-free deduplication
- Composite key: (task_id, scheduled_time, store_id) ensures proper isolation
- Cleanup functionality to prevent database growth

### Task Execution  
- Integrates with existing AgenticLoop for LLM execution
- Proper error handling with failed task events
- Event emission for task lifecycle (start, complete, fail)
- Configurable execution parameters and context

### Render Cron Integration
- Standalone script that can run independently
- Clean separation from main server process  
- Proper environment validation and graceful shutdown
- Statistics reporting and resource cleanup

## Test Plan

✅ All 33 tests passing:
- ProcessedTaskTracker: 15 tests (database, deduplication, concurrency)
- TaskScheduler: 13 tests (execution, error handling, windowing)  
- process-tasks script: 5 tests (main function, error cases)

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: packages/server/src/services/task-scheduler.ts [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+    const completedExecutions = store.query(
+      queryDb(
+        tables.taskExecutions
+          .select()
+          .where('recurringTaskId', '=', taskId)
+          .where('status', '=', 'completed')
+          .where('startedAt', '>=', windowStart)
+          .where('startedAt', '<=', windowEnd)
+      )
+    ) as TaskExecution[]
```
Comment (jessmartin) [inline packages/server/src/services/task-scheduler.ts:295]: This code looks suspicious. Why would I need to query using sql directly?
